### ISSUE TITLE
[Bug]: About Window Shows Wrong Version of SQLite.dll

### ISSUE THREAD

**Issue by NoralK** [sqlitebrowser/sqlitebrowser#3135]
### What did you do?

Dropped in latest x64 SQLite.dll from https://www.sqlite.org/download.html.

### What did you expect to see?

Should be showing 3.39.2 in the About window.

### What did you see instead?

"About DB Browser for SQLite" still shows 3.35.5 but when I do `select sqlite_version();` it shows 3.39.2.

### DB4S Version

3.12.2

### What OS are you seeing the problem on?

Windows

### OS version

Windows 10 - 21H2

### Relevant log output

```shell
N/A
```

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by lucydodo** [comment]
The SQLite version output by DB4S is determined at compile time, not at execution time. (see code below)
https://github.com/sqlitebrowser/sqlitebrowser/blob/a78dd0292f7e1149b4e81a3987531aa67098b92c/src/sqlitedb.cpp#L543
>The constant SQLITE_VERSION value is in your installed SQLite library code (`sqlite3.h`).

Personally, I think replacing the DLL file is a kind of tampering concept,
whether it's a good idea to determine the SQLite version at execution time is a topic worth pondering.

@MKleusberg What do you think about this issue? 🤔

**Comment by mgrojo** [comment]
Maybe we should include both versions. The version compiled was the one we tested and delivered. The execution time version is the one actually being used by the application, with its advantages (you could put a newer version) and disadvantages (you could get bugs unknown to us).

**Comment by lucydodo** [comment]
>Maybe we should include both versions.

This is a really great idea.

**Comment by scottfurry** [comment]
And now we have confusion...
SQLite is prolific on multiple platforms. I have yet to encounter where a distribution _did not_ install some version. So, are we packaging sqlite or using the system installed version? 

This dovetails into another discussion (see #3419 - Item: Update Dependent Libraries and Document Update Methodology) about how best to maintain and document dependencies. SQLite version should be another item but with special care given it's placement in the source tree.

What appears on the about dialog _should be_ the version included with the app. The user should be able to determine the version on their own system. An FAQ item written to explain if any discrepancy (e.g. Q:"App shows a different SQLite version?" A:"SQLite version in DB4S source and not version installed on system").

I agree with the observation above about replacing dynamic library objects is not a good idea.

tl;dr - SQLIte is a project dependency and we should manage and convey version to user better.

**Comment by chrisjlocke** [comment]
> I have yet to encounter where a distribution did not install some version.

Its not on Windows.  
But by 'distribution', I assume you mean Linux distribution...

Again, this is where Windows and Linux differ slightly.  While we bundle SQLite with DB4S, Windows will use that version first, while on Linux, if its installed on the 'system', then that version is used, ignoring the bundled version (I think that's right?)
This recently popped up in a support issue where someone was using a nightly build of SQLCipher, but SQLite was 3.0 or something...

**Comment by chrisjlocke** [comment]
> The execution time version is the one actually being used by the application

Is there a way to determine where this version lives?  I know there is 'which' but thats a third-party application.  Wondering if it was a simple way of determining, 'the version of sqlite you're running lives in this folder' which can help determine, 'aah, it was installed by that program'.

**Comment by scottfurry** [comment]
POSIX platform (BSD, Linux, et al). However, I see your point @chrisjlocke. Windows is a different _beast_.

When executing, the application will make the call to the system to access sqlite via its library. Where it is found in the system $PATH is an entirely different story. My assumption is that the executable path will be searched before other paths. So if the DB4S sits in it's own directory under `C:\Win32` et al, that sqlite object library will be accessed before the system - in theory.

If SQLite was installed separately, Windows mechanism will show that install version of the dll.

I liken this discussion to one of _separation of concern_. Are we, or should we, be concerned with the version of SQLite the user _might have_ installed? Nope. Are we concerned with the SQLite version we include? Absolutely.

**Comment by scottfurry** [comment]
> Is there a way to determine where this version lives? I know there is 'which' but thats a third-party application. Wondering if it was a simple way of determining, 'the version of sqlite you're running lives in this folder' which can help determine, 'aah, it was installed by that program'.

`which` returns a single path and is available on any POSIX platform.

`whereis` also returns stuff but this becomes a multi-step answer:
`$ whereis sqlite3`
parse for binary path (e.g. `sqlite3: /usr/bin/sqlite3 /usr/share/man/man1/sqlite3.1.bz2`)
`sqlite3 -version`
parse return if successful
output: `3.43.1 2023-09-11 12:01:27 2d3a40c05c49e1a49264912b1a05bc2143ac0e7c3df588276ce80a4cbc9balt1 (64-bit)`

Again, separation-of-concern. Do we need to go this extra step?

**Comment by chrisjlocke** [comment]
> Are we, or should we, be concerned with the version of SQLite the user might have installed? Nope.

Only from a support point of view.  We get a couple of issues with people confused they've downloaded version X of DB4S which should have version Y of SQLite but the 'strict' keyword isn't working, for example.  If DB4S clearly showed it was using v3.0.00.0.0...0000 of SQLite, it would be helpful.

**Comment by scottfurry** [comment]
...which will confuse the user about which sqlite that is being used by the application.

I can see managing where system sqlite is less than DB4S sqlite. And if we're looking for that... _scope creep_

I'm surprised we haven't statically linked in sqlite3 into the app - the question then becomes mute.

Again, this is a _dependency problem_ for us. Staying on top of SQLite3 versioning is important.
